src/hal/utils/halcmd_commands.c

```diff
@@ -1975,58 +1975,58 @@ static int print_comp_entry(hal_object_ptr o, foreach_args_t *args)
-		      type_name(comp),
-		      has_ctor ? 'c': ' ',
-		      has_dtor ? 'd': ' ',
-		      is_hallib ? 'i': ' ',
-		      ' ',
-		      inst_count(0, comp),
-		      HAL_NAME_LEN,
-		      ho_name(comp));
-
-	switch (comp->type) {
-	case TYPE_USER:
-	case TYPE_HALLIB:
-
-	    halcmd_output(" %-5d %s", comp->pid,
-			  state_name(comp->state));
-	    break;
-
-	case TYPE_RT:
-	    halcmd_output(" RT    %s",
-			  state_name(comp->state));
-	    break;
-
-	case TYPE_REMOTE:
-	    halcmd_output(" %-5d %s", comp->pid,
-			  state_name(comp->state));
-	    time_t now = time(NULL);
-	    if (comp->last_update) {
-
-		halcmd_output(", update:-%ld",-(comp->last_update-now));
-	    } else
-		halcmd_output(", update:never");
-
-	    if (comp->last_bound) {
-
-		halcmd_output(", bound:%lds",comp->last_bound-now);
-	    } else
-		halcmd_output(", bound:never");
-	    if (comp->last_unbound) {
-		time_t now = time(NULL);
-
-		halcmd_output(", unbound:%lds", comp->last_unbound-now);
-	    } else
-		halcmd_output(", unbound:never");
-		halcmd_output(", u1:%d u2:%d", comp->userarg1, comp->userarg2);
-	    break;
-	default:
-	    halcmd_output(" %-5s %s", "", state_name(comp->state));
-	}
-	halcmd_output(", u1:%d u2:%d", comp->userarg1, comp->userarg2);
-	halcmd_output("\n");
+        halcmd_output(" %5d  %-4s %c%c%c%c  %4d %-*s",
+                      ho_id(comp),
+                      type_name(comp),
+                      has_ctor ? 'c': ' ',
+                      has_dtor ? 'd': ' ',
+                      is_hallib ? 'i': ' ',
+                      ' ',
+                      inst_count(0, comp),
+                      HAL_NAME_LEN,
+                      ho_name(comp));
+
+        switch (comp->type) {
+            case TYPE_USER:
+            case TYPE_HALLIB:
+
+                halcmd_output(" %-5d %s", comp->pid,
+                              state_name(comp->state));
+                break;
+
+            case TYPE_RT:
+                halcmd_output(" RT    %s",
+                              state_name(comp->state));
+                break;
+
+            case TYPE_REMOTE:
+                halcmd_output(" %-5d %s", comp->pid,
+                              state_name(comp->state));
+                time_t now = time(NULL);
+                if (comp->last_update) {
+
+                    halcmd_output(", update:-%ld",-(comp->last_update-now));
+                } else
+                    halcmd_output(", update:never");
+
+                if (comp->last_bound) {
+
+                    halcmd_output(", bound:%lds",comp->last_bound-now);
+                } else
+                    halcmd_output(", bound:never");
+                if (comp->last_unbound) {
+                    time_t now = time(NULL);
+
+                    halcmd_output(", unbound:%lds", comp->last_unbound-now);
+                } else
+                    halcmd_output(", unbound:never");
+                halcmd_output(", u1:%d u2:%d", comp->userarg1, comp->userarg2);
```

Comment:
I have looked at this before and am far from convinced that this line should be here at all, never mind indented.

After the halcmd_output(",unbound:...............) line from either if/else conditional, execution will exit the switch() and go to exactly the same line at 2028.
So if `comp->last_unbound == NULL` it will print the same line twice

Reply:
Hrm, honestly I wasn't reading the code carefully, just fixing errors!

I don't have time to look now, gotta get some homemade dumplings while they're still hot.  If you leave this PR open another day or two, I'll try to take a look at it.